cartridge.lua

```diff
@@ -593,12 +593,14 @@ local function cfg(opts, box_opts)
     local auth_backend, err = CartridgeCfgError:pcall(require, opts.auth_backend_name)
     if not auth_backend then
         return nil, err
+    elseif type(auth_backend) ~= 'table' then
+        return nil, CartridgeCfgError:new(
+            "Auth backend must export a table, got %s",
+            type(auth_backend)
+        )
     end
 
-    local ok, err = CartridgeCfgError:pcall(function()
-        local ok = auth.set_callbacks(auth_backend)
-        return ok
```

Comment:
Historical remark. One could ask: why do we need these `local ok, return ok`, why not simply call `return set_callbacks()`? The reason was in the `checks` which spoiled the message. Simple `return` drains a frame from the call stack, leaving `pcall` on the top. And the error looks as follows:

```log
bad argument callback.check_password to nil
```

This patch eliminates the generation of error messages based on `checks`, so we can rewrite this code in a more straightforward way.